icu4c/source/test/intltest/collationtest.cpp

```diff
@@ -1754,10 +1754,10 @@ UBool CollationTest::checkCompareTwo(const char *norm, const UnicodeString &prev
     //   sortkey(str1 + "\uFFFE" + str2) == mergeSortkeys(sortkey(str1), sortkey(str2))
     // only that those two methods yield the same order.
     //
-    // Use bit-wise OR so that getMergedCollationKey() is always called for both strings.
-    if((getMergedCollationKey(prevString.getBuffer(), prevString.length(), prevKey, errorCode) |
-                getMergedCollationKey(s.getBuffer(), s.length(), key, errorCode)) ||
-            errorCode.isFailure()) {
+    // Use two variables so that getMergedCollationKey() is always called for both strings.
+    if (UBool prev = getMergedCollationKey(prevString.getBuffer(), prevString.length(), prevKey, errorCode),
+              curr = getMergedCollationKey(s.getBuffer(), s.length(), key, errorCode);
+        prev || curr || errorCode.isFailure()) {
```

Comment:
This seems like it'd read better if we just moved the definitions of `prev` and `curr ` out of the condition of the `if` statement.

Reply:
optional: I agree that if we can't use the bit-wise OR any more then pulling out the two variables, like in the other file, is more readable. Mostly because of the fairly long function calls.

Reply:
Maybe, but I don't think that's something that'd be worth spending another round of review/approval on.